### PR TITLE
Update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by adidalal**
Change examples to show versioned URLs more prominently and add in further explanation about squashing commits.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -28,39 +28,46 @@ Making a Cask is easy: a Cask is a small Ruby file.
 
 ### Examples
 
-Here’s a Cask for `Alfred.app` as an example. Note that you may repeat the `app` stanza as many times as you need, to define multiple apps:
+Here’s a Cask for `Shuttle.app` as an example. Note that the `url` stanza uses `#{version}` (string interpolation) to create a Cask that only needs `version` and `sha256` updated with an application update.
```
Comment (vitorgalvao) [inline CONTRIBUTING.md]: Suggestion so we don’t repeat “update”: “that only needs `version` and `sha256` changes when updated.”

Comment (vitorgalvao) [inline CONTRIBUTING.md]: I’d also suggest for “string interpolation” to be a link to https://en.wikipedia.org/wiki/String_interpolation#Ruby.

Comment (vitorgalvao) [inline CONTRIBUTING.md]: Also, `shuttle`, instead of `Shuttle.app`, to be consistent with the next example.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -381,7 +388,7 @@ Congratulations! You are done now, and your Cask should be pulled in or otherwis
 
 ### Squashing
 
-If your pull request has multiple commits which revise the same lines of code, it is better to [squash](http://davidwalsh.name/squash-commits-git) those commits together into one logical unit.
+If your pull request has multiple commits which revise the same lines of code, or if you make some changes after comments from one of the maintainers, it is better to [squash](http://davidwalsh.name/squash-commits-git) those commits together into one logical unit.
```
Comment (vitorgalvao) [inline CONTRIBUTING.md]: Big thumb up to making it explicit here.

**Comment by adityadalal924** [comment]
Noticed that firefox isn't actually unversioned anymore, so I changed it to another example with an appropriate url (and one with a comment, just so that there's a clear example on the page)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -371,7 +381,7 @@ $ github_user='<my-github-username>'
 $ git push "$github_user" my-new-cask
 ```
 
-If you are using [GitHub two-factor authentication](https://github.com/blog/1614-two-factor-authentication) and set your remote repository as HTTPS you will need to set up a personal access token and use that instead your password. See more on https://help.github.com/articles/https-cloning-errors#provide-access-token-if-2fa-enabled
+If you are using [GitHub two-factor authentication](https://help.github.com/articles/about-two-factor-authentication/) and set your remote repository as HTTPS you will need to set up a personal access token and use that instead your password. See more on [here](https://help.github.com/articles/https-cloning-errors/#provide-access-token-if-2fa-enabled).
```
Comment (vitorgalvao) [inline CONTRIBUTING.md]: “access token and use that instead **of** your password”.

Comment (vitorgalvao) [inline CONTRIBUTING.md]: “See more here” (no “on”).

**Comment by adityadalal924** [comment]
Merged as 41aa133fe24fd7705dd71626a7506dab4db2c3bf.

**Pull request closed without merging**